Updated api version of providerhub cli to 2024-04-01-preview

gaurimadhok opened the pull request:
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
<!--- Please provide the related command with az {command} if you can, so that we can quickly route to the related person to review. --->
az providerhub

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).


### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

yonzhan commented:
<!-- add-pr-comment:prcommentbot -->

Thank you for your contribution! We will review the pull request and get back to you soon.

kairu-ms commented on src/providerhub/azext_providerhub/generated/commands.py:
```diff
@@ -15,103 +15,9 @@
         operations_tmpl='azext_providerhub.vendored_sdks.providerhub.operations._skus_operations#SkusOperations.{}',
         client_factory=cf_sku)
     with self.command_group('providerhub sku', providerhub_sku, client_factory=cf_sku) as g:
         g.custom_command('list', 'providerhub_sku_list')
```
It seems the list command is generated, why still keep the old command?

gaurimadhok commented:
> ❌Azure CLI Extensions Breaking Change Test
> > ❌providerhub
> > 
> > 
> > 
> > rule
> > cmd_name
> > rule_message
> > suggest_message
> > 
> > 
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > providerhub default-rollout create
> > cmd `providerhub default-rollout create` update parameter `rest_of_the_world_group_two`: updated property `options` from `['--rest-of-the-world-group-two', '--row2']` to `['--rest-of-the-world-group-two']`
> > please change property `options` from `['--rest-of-the-world-group-two']` to `['--rest-of-the-world-group-two', '--row2']` for parameter `rest_of_the_world_group_two` of cmd `providerhub default-rollout create`
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > providerhub manifest checkin
> > cmd `providerhub manifest checkin` update parameter `baseline_arm_manifest_location`: updated property `options` from `['--arm-manifest-location', '--baseline-arm-manifest-location', '--location']` to `['--baseline-arm-manifest-location']`
> > please change property `options` from `['--baseline-arm-manifest-location']` to `['--arm-manifest-location', '--baseline-arm-manifest-location', '--location']` for parameter `baseline_arm_manifest_location` of cmd `providerhub manifest checkin`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` removed parameter `lighthouse_authorizations`
> > please add back parameter `lighthouse_authorizations` for cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` removed parameter `managed_by_tenant_id`
> > please add back parameter `managed_by_tenant_id` for cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` removed parameter `providerhub_metadata_authentication`
> > please add back parameter `providerhub_metadata_authentication` for cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` removed parameter `providerhub_metadata_authorizations`
> > please add back parameter `providerhub_metadata_authorizations` for cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` update parameter `subscription_state_override_actions`: updated property `options` from `['--override-actions', '--subscription-state-override-actions']` to `['--subscription-state-override-actions']`
> > please change property `options` from `['--subscription-state-override-actions']` to `['--override-actions', '--subscription-state-override-actions']` for parameter `subscription_state_override_actions` of cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > providerhub provider-registration create
> > cmd `providerhub provider-registration create` update parameter `template_deployment_options`: updated property `options` from `['--template-deployment-options', '--tmplt-deploy-opt']` to `['--template-deployment-options']`
> > please change property `options` from `['--template-deployment-options']` to `['--template-deployment-options', '--tmplt-deploy-opt']` for parameter `template_deployment_options` of cmd `providerhub provider-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub resource-type-registration create
> > cmd `providerhub resource-type-registration create` removed parameter `enable_third_party_s2s`
> > please add back parameter `enable_third_party_s2s` for cmd `providerhub resource-type-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub resource-type-registration create
> > cmd `providerhub resource-type-registration create` removed parameter `resource_patch_begin`
> > please add back parameter `resource_patch_begin` for cmd `providerhub resource-type-registration create`
> > 
> > 
> > ❌ [1010 - ParaPropUpdate](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1010.md)
> > providerhub resource-type-registration create
> > cmd `providerhub resource-type-registration create` update parameter `subscription_state_override_actions`: updated property `options` from `['--override-actions', '--subscription-state-override-actions']` to `['--subscription-state-override-actions']`
> > please change property `options` from `['--subscription-state-override-actions']` to `['--override-actions', '--subscription-state-override-actions']` for parameter `subscription_state_override_actions` of cmd `providerhub resource-type-registration create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku create
> > cmd `providerhub sku create` removed parameter `nested_resource_type_first`
> > please add back parameter `nested_resource_type_first` for cmd `providerhub sku create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku create
> > cmd `providerhub sku create` removed parameter `nested_resource_type_second`
> > please add back parameter `nested_resource_type_second` for cmd `providerhub sku create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku create
> > cmd `providerhub sku create` removed parameter `nested_resource_type_third`
> > please add back parameter `nested_resource_type_third` for cmd `providerhub sku create`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku delete
> > cmd `providerhub sku delete` removed parameter `nested_resource_type_first`
> > please add back parameter `nested_resource_type_first` for cmd `providerhub sku delete`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku delete
> > cmd `providerhub sku delete` removed parameter `nested_resource_type_second`
> > please add back parameter `nested_resource_type_second` for cmd `providerhub sku delete`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku delete
> > cmd `providerhub sku delete` removed parameter `nested_resource_type_third`
> > please add back parameter `nested_resource_type_third` for cmd `providerhub sku delete`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku list
> > cmd `providerhub sku list` removed parameter `nested_resource_type_first`
> > please add back parameter `nested_resource_type_first` for cmd `providerhub sku list`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku list
> > cmd `providerhub sku list` removed parameter `nested_resource_type_second`
> > please add back parameter `nested_resource_type_second` for cmd `providerhub sku list`
> > 
> > 
> > ❌ [1007 - ParaRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1007.md)
> > providerhub sku list
> > cmd `providerhub sku list` removed parameter `nested_resource_type_third`
> > please add back parameter `nested_resource_type_third` for cmd `providerhub sku list`
> > 
> > 
> > ❌ [1002 - CmdRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1002.md)
> > providerhub sku show-nested-resource-type-first
> > cmd `providerhub sku show-nested-resource-type-first` removed
> > please confirm cmd `providerhub sku show-nested-resource-type-first` removed
> > 
> > 
> > ❌ [1002 - CmdRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1002.md)
> > providerhub sku show-nested-resource-type-second
> > cmd `providerhub sku show-nested-resource-type-second` removed
> > please confirm cmd `providerhub sku show-nested-resource-type-second` removed
> > 
> > 
> > ❌ [1002 - CmdRemove](https://github.com/Azure/azure-cli/blob/dev/doc/breaking_change_rules/1002.md)
> > providerhub sku show-nested-resource-type-third
> > cmd `providerhub sku show-nested-resource-type-third` removed
> > please confirm cmd `providerhub sku show-nested-resource-type-third` removed

I have reviewed these and updated the ones that needed to be updated.

yonzhan commented:
Please fix CI issues

kairu-ms commented:
/azp run
